stockflux-launcher/src/free-text-search/FreeTextSearch.js

```diff
@@ -0,0 +1,216 @@
+import React, {
+  useState,
+  useReducer,
+  useEffect,
+  useRef,
+  useCallback
+} from 'react';
+import { FaSearch } from 'react-icons/fa';
+import { StockFlux } from 'stockflux-core';
+import SearchResult from './search-result';
+import {
+  InterApplicationBusHooks,
+  WindowHooks,
+  Constants
+} from 'openfin-react-hooks';
+import { createWindow, populateResultsContainer } from './SearchResultWindow';
+import './FreeTextSearch.css';
+
+const SEARCHING = 'searching';
+const SUCCESS = 'success';
+const ERROR = 'error';
+const INITIALISE = 'initialise';
+const SEARCH_TIMEOUT_INTERVAL = 250;
+const NO_MATCHES_HTML = <p>Sorry, no matches found.</p>;
+const INITIAL_MESSAGE_HTML = (
+  <p>
+    Use the input field above to search for instruments or click on the search
+    icon to close the search.
+  </p>
+);
+
+let latestRequest = null;
+let resultsWindow = null;
+
+const initialSearchState = {
+  isSearching: false,
+  hasErrors: false
+};
+
+const searchReducer = (state, { type, results }) => {
+  switch (type) {
+    case SEARCHING:
```

Comment:
Could we have a loading spinner being shown when fetching results?

Reply:
I think it would be easier to wait while this branch gets merged into `feature-launcher` branch and then rebase that on top of `cra-wip` which will contain spinner component inside `stockflux-components`.